test/api/mm/tests

```diff
@@ -0,0 +1,8 @@
+extcharges
```

Comment:
Would be good to have 
a) a header
b) C equivalents to all of the tests (presumably people may use these as templates, especially if point to in the documentation).